datasets/flwr_datasets/visualization/label_distribution.py

```diff
@@ -0,0 +1,643 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Label distribution plotting."""
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+import datasets
+from flwr_datasets.metrics import compute_counts
+from flwr_datasets.partitioner import Partitioner
+
+# pylint: disable=too-many-arguments,too-many-locals
+# Constants for plot types and size units
+PLOT_TYPES = ("bar", "heatmap")
+SIZE_UNITS = ("absolute", "percent")
+AXIS_TYPES = ("x", "y")
+
+
+def plot_label_distributions(
+    partitioner: Partitioner,
+    label_name: str,
+    plot_type: str = "bar",
+    size_unit: str = "absolute",
+    max_num_partitions: Optional[int] = None,
+    partition_id_axis: str = "x",
+    axis: Optional[Axes] = None,
+    figsize: Optional[Tuple[float, float]] = None,
+    title: str = "Per Partition Label Distribution",
+    cmap: Optional[Union[str, mcolors.Colormap]] = None,
+    legend: bool = False,
+    legend_title: str = "Labels",
+    verbose_labels: bool = True,
+    **plot_kwargs: Dict[str, Any],
```

Comment:
```suggestion
    **plot_kwargs: Dict[str, Any],
    **legend_kwargs: Dict[str, Any],
```

Reply:
I'd suggest also allowing for passing arguments for the legend. I can see users wanting to specify things like `ncols`, `loc` `bbox_to_anchor` at least. From my exp, there is not single combination that fits well all plots (primarily because of sizes of figures)

Reply:
Ok, I agree. Though two `**kwargs` are techniqually impossible, we can make both of them but drop the `**`. In that case user will need to give them in dict (I'd drop the `**` from both in that case), then we can have 
```
    plot_kwargs: Dict[str, Any],
    legend_kwargs: Dict[str, Any],
 ```
Also, when using that we need and handle the overlap (e.g. someone gives the legend_name argument and also specifies the legend title in the legend_kwargs). I think that the explicit arguments should take precedence (e.g. the legend name specified as legend_name not in the kwargs). Wdyt?